src/components/App.js

```diff
@@ -1,21 +1,32 @@
-import React, { PureComponent } from 'react';
+import React, { Component } from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+
 import calculate from '../logic/calculate';
 
-class App extends PureComponent {
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: '0',
+      next: '0',
+      operation: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick= btnName => {
+    const result = calculate(this.state, btnName);
+    this.setState(result);
+  }
```

Comment:
__[Optional]__ Implementing `handleClick` as an arrow function and then binding it in the constructor is redundant since arrow functions don't require this binding explicitly in classes.